assembly/src/tokens/stream.rs

```diff
@@ -26,18 +27,23 @@ impl<'a> TokenStream<'a> {
         }
         let mut tokens = Vec::new();
         let mut doc_comments = BTreeMap::new();
+        let mut module_comment = None;
 
-        let mut comment = Comment(None);
+        let mut doc_comment = Comment(None);
 
         for line in source.lines() {
             let line = line.trim();
             if line.starts_with(DOC_COMMENT_PREFIX) {
-                comment.append_line(line);
-            } else if line.is_empty() && !comment.is_empty() {
-                return Err(ParsingError::malformed_doc_comment(tokens.len()));
+                doc_comment.append_line(line);
+            } else if line.is_empty() && !doc_comment.is_empty() {
+                if tokens.is_empty() && module_comment.is_none() {
+                    module_comment = doc_comment.take_content();
+                } else {
+                    return Err(ParsingError::malformed_doc_comment(tokens.len()));
+                }
```

Comment:
I wonder if we can get rid of this check and instead have a check in the "else" clause to make sure that a procedure comment is always followed by a procedure token (i.e., the one that starts with `export` or `proc`).